### ISSUE TITLE
FTP upload messes file permissions on first upload after long idle

### ISSUE THREAD

**Issue by Kanimies** [eranif/codelite#1713]
When a file has been open in the editor for a long period of time without being touched - and Codelite being left running as well, the first time the file is saved the automatic FTP upload does transfer the file but fails to set the file permissions - they are left to none. A consecutive edit + save and thus automatic  upload again fixes this issue. 

The bug only seems to appear after the file being left untampered for several hours, but can be repeated every time. SFTP Log always logs a successful upload.

Platform: Mac OS X, versions since last spring at least up until 10.0.6.

**Comment by Kanimies** [comment]
Also seems that the automatic FTP does these uploads when simply opening a file for editing even when no changes are made to the file - and again messes up the file permissions on the server end, but not every time.

**Comment by Kanimies** [comment]
This is what the sftp system log seems to look like for one such erroneous transfer
---
Aug 11 06:56:08 hostname sftp-server[29330]: session opened for local user nobody from [192.168.0.260]
Aug 11 06:56:08 hostname sftp-server[29330]: received client version 3
Aug 11 06:56:08 hostname sftp-server[29330]: stat name "/path"
Aug 11 06:56:08 hostname sftp-server[29330]: debug1: request 16777216: sent attrib have 0xf
Aug 11 06:56:08 hostname sftp-server[29330]: stat name "/path/to"
Aug 11 06:56:08 hostname sftp-server[29330]: debug1: request 33554432: sent attrib have 0xf
Aug 11 06:56:08 hostname sftp-server[29330]: stat name "/path/to/file"
Aug 11 06:56:08 hostname sftp-server[29330]: debug1: request 50331648: sent attrib have 0xf
Aug 11 06:56:08 hostname sftp-server[29330]: stat name "/path/to/file/directory"
Aug 11 06:56:08 hostname sftp-server[29330]: debug1: request 67108864: sent attrib have 0xf
Aug 11 06:56:08 hostname sftp-server[29330]: stat name "/path/to/file/directory/name"
Aug 11 06:56:08 hostname sftp-server[29330]: debug1: request 83886080: sent attrib have 0xf
Aug 11 06:56:08 hostname sftp-server[29330]: stat name "/path/to/file/directory/name/here"
Aug 11 06:56:08 hostname sftp-server[29330]: debug1: request 100663296: sent attrib have 0xf
Aug 11 06:56:08 hostname sftp-server[29330]: open "/path/to/file/directory/name/here/filename.php.codelitesftp" flags WRITE,CREATE,TRUNCATE mode 0644
Aug 11 06:56:08 hostname sftp-server[29330]: debug1: request 117440512: sent handle handle 0
Aug 11 06:56:08 hostname sftp-server[29330]: debug1: request 134217728: write "/path/to/file/directory/name/here/filename.php.codelitesftp" (handle 0) off 0 len 247
Aug 11 06:56:08 hostname sftp-server[29330]: sent status Success
Aug 11 06:56:08 hostname sftp-server[29330]: close "/path/to/file/directory/name/here/filename.php.codelitesftp" bytes read 0 written 247
Aug 11 06:56:08 hostname sftp-server[29330]: sent status Success
Aug 11 06:56:08 hostname sftp-server[29330]: stat name "/path/to/file/directory/name/here/filename.php"
Aug 11 06:56:08 hostname sftp-server[29330]: debug1: request 167772160: sent attrib have 0xf
Aug 11 06:56:08 hostname sftp-server[29330]: remove name "/path/to/file/directory/name/here/filename.php"
Aug 11 06:56:08 hostname sftp-server[29330]: sent status Success
Aug 11 06:56:08 hostname sftp-server[29330]: rename old "/path/to/file/directory/name/here/filename.php.codelitesftp" new "/path/to/file/directory/name/here/filename.php"
Aug 11 06:56:08 hostname sftp-server[29330]: sent status Success
Aug 11 06:56:08 hostname sftp-server[29330]: debug1: request 218103808: setstat name "/path/to/file/directory/name/here/filename.php"
Aug 11 06:56:08 hostname sftp-server[29330]: set "/path/to/file/directory/name/here/filename.php" **mode 0162**
Aug 11 06:56:08 hostname sftp-server[29330]: sent status Success

---

The mode 0162 strikes into my eye - why is that being set?

**Comment by Kanimies** [comment]
Another one just now that set the permissions to none - again after a file as saved after being open for a longer period of time. The SFTP connection timed out first - could've been a connection left open by the program earlier.

Also the program keeps overwriting files on the server upon opening them, not just on save, which tends to lead to problems with multiple users using the feature. What's the reasoning behind transferring the file to remote end with automatic SFTP upload on opening anyway?
```

Aug 21 13:37:35 servername sftp-server[30623]: stat name "/path1"
Aug 21 13:37:35 servername sftp-server[30623]: debug1: request 167772160: sent attrib have 0xf
Aug 21 13:37:35 servername sftp-server[30623]: stat name "/path1/path2"
Aug 21 13:37:35 servername sftp-server[30623]: debug1: request 184549376: sent attrib have 0xf
Aug 21 13:37:35 servername sftp-server[30623]: stat name "/path1/path2/path3"
Aug 21 13:37:35 servername sftp-server[30623]: debug1: request 201326592: sent attrib have 0xf
Aug 21 13:37:35 servername sftp-server[30623]: open "/path1/path2/path3/filename.php.codelitesftp" flags WRITE,CREATE,TRUNCATE mode 0644
Aug 21 13:37:35 servername sftp-server[30623]: debug1: request 218103808: sent handle handle 0
Aug 21 13:37:35 servername sftp-server[30623]: debug1: request 234881024: write "/path1/path2/path3/filename.php.codelitesftp" (handle 0) off 0 len 14959
Aug 21 13:37:35 servername sftp-server[30623]: sent status Success
Aug 21 13:37:35 servername sftp-server[30623]: close "/path1/path2/path3/filename.php.codelitesftp" bytes read 0 written 14959
Aug 21 13:37:35 servername sftp-server[30623]: sent status Success
Aug 21 13:37:35 servername sftp-server[30623]: stat name "/path1/path2/path3/filename.php"
Aug 21 13:37:35 servername sftp-server[30623]: debug1: request 268435456: sent attrib have 0xf
Aug 21 13:37:35 servername sftp-server[30623]: remove name "/path1/path2/path3/filename.php"
Aug 21 13:37:35 servername sftp-server[30623]: sent status Success
Aug 21 13:37:35 servername sftp-server[30623]: rename old "/path1/path2/path3/filename.php.codelitesftp" new "/path1/path2/path3/filename.php"
Aug 21 13:37:35 servername sftp-server[30623]: sent status Success
Aug 21 13:37:35 servername sftp-server[30623]: debug1: request 318767104: setstat name "/path1/path2/path3/filename.php"
Aug 21 13:37:35 servername sftp-server[30623]: set "/path1/path2/path3/filename.php" mode 0000
Aug 21 13:37:35 servername sftp-server[30623]: sent status Success
```

**Comment by eranif** [comment]
> Also seems that the automatic FTP does these uploads when simply opening a file for editing even when no changes are made to the file - and again messes up the file permissions on the server end, but not every time.

This was already fixed in this commit 1f33a21

**Comment by eranif** [comment]
I will be releasing CodeLite 10.0.7 this weekend it should include at least the "Save on open" bug
